sound/soc/intel/boards/sof_sdw.c

```diff
@@ -323,7 +439,29 @@ static int first_spk_init(struct snd_soc_pcm_runtime *rtd)
 	struct snd_soc_card *card = rtd->card;
 	int ret;
 
-	ret = snd_soc_dapm_add_routes(&card->dapm, rt1308_speaker_map, 2);
+	ret = snd_soc_add_card_controls(card, rt1308_controls,
+					ARRAY_SIZE(rt1308_controls));
+	if (ret) {
+		dev_err(card->dev, "rt1308 controls addition failed: %d\n", ret);
+		return ret;
+	}
+
+	ret = snd_soc_dapm_new_controls(&card->dapm, rt1308_widgets,
+					ARRAY_SIZE(rt1308_widgets));
+	if (ret) {
+		dev_err(card->dev, "rt1308 widgets addition failed: %d\n", ret);
+		return ret;
+	}
+
+	ret = snd_soc_dapm_add_routes(&card->dapm, rt1308_map,
+				      ARRAY_SIZE(rt1308_map));
+
+	if (ret)
+		dev_err(card->dev, "rt1308 map addition failed: %d\n", ret);
+
+	return ret;
```

Comment:
merge error ?   duplicate  snd_soc_dapm_add_routes.  For above case, error is:
````
[    3.986730] sof_sdw sof_sdw: ASoC: no source widget found for rt1308-2 SPOL
[    3.986732] sof_sdw sof_sdw: ASoC: Failed to add route rt1308-2 SPOL -> direct -> Speaker
[    3.986736] sof_sdw sof_sdw: ASoC: no source widget found for rt1308-2 SPOR
[    3.986737] sof_sdw sof_sdw: ASoC: Failed to add route rt1308-2 SPOR -> direct -> Speaker
[    3.986739] sof_sdw sof_sdw: rt1308 map addition failed: -19
[    3.986741] sof_sdw sof_sdw: ASoC: failed to init SDW1-Playback: -19
 ````

Reply:
Fixed now, bad merge with copy/paste. Can you give it a try?